src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -358,9 +359,14 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
       // Create the cluster
       machineConfig = MachineConfigOps.create(clusterRequest.machineConfig, clusterDefaultsConfig)
       initScript = GcsPath(initBucket, GcsObjectName(clusterResourcesConfig.initActionsScript.value))
+      autopauseThreshold = clusterRequest.autopause match {
+        case None => autoFreezeConfig.autoFreezeAfter
+        case Some(false) => 0
+        case _ => if (clusterRequest.autopauseThreshold == None) autoFreezeConfig.autoFreezeAfter else clusterRequest.autopauseThreshold.get
```

Comment:
Can we sanitize against other potential bad inputs? What if we get:
```
autopause = Some(true)
autopauseThreshold = Some(-2452345)
```
Currently we'd pass this directly to the DB query and bad things might happen. `Math.max(0, autopauseThreshold)` would probably resolve this particular issue.

Otherwise, the logic looks good, I think.